pkg/streamingpromql/functions/rate.go

```diff
@@ -95,6 +100,10 @@ func histogramRate(histogramBuffer *types.HPointRingBuffer, step types.RangeVect
 				emitAnnotation(annotations.NewNativeHistogramNotCounterWarning)
 			}
 
+			if p.H.UsesCustomBuckets() != usingCustomBuckets {
```

Comment:
Why not do this check first thing in this for? It could potentially avoid doing an unnecessary CopyToSchema

Reply:
Changed in 1558d29